test/build/50_localregistry.bats

```diff
@@ -40,6 +42,41 @@ EOF
     #[[ $output =~ $re ]]
 }
 
+
+@test "${tag}: with upload cache" {
+    [[ $CH_IMAGE_CACHE != disabled ]] || skip 'build cache disabled'
+    run ch-image upload-cache --reset
+    echo "$output"
+    [[ $status -eq 0 ]]
+    [[ $output = *'upload files:       0'* ]]
+
+    # Push and store prepared upload files.
+    ch-image -v --tls-no-verify push --ulcache alpine:3.17 localhost:5000/ulcache
+    run ch-image upload-cache
+    echo "$output"
+    [[ $status -eq 0 ]]
+    [[ $output = *'upload files:       3'* ]]
+
+    # Reuse previously prepared files
+    run ch-image -v --tls-no-verify push --ulcache alpine:3.17 localhost:5000/ulcache:prepared
+    echo "output"
+    [[ $status -eq 0 ]]
+    [[ $output = *'using previously prepared config'* ]]
+    [[ $output = *'using previously prepared manifest'* ]]
+    [[ $output = *'using previously prepared layer(s)'* ]]
+}
+
+
+@test "${tag}: upload cache error(s)" {
+    [[ $CH_IMAGE_CACHE == disabled ]] || skip 'build cache enabled'
```

Comment:
I'd gate this test with `[[ $CH_IMAGE_CACHE == enabled ]]`, then use `--no-cache` below. As is, I think it sets up a gotcha if there is more added to this function that doesn’t require the cache to be disabled.